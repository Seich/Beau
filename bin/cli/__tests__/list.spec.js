const ListCommand = require('../commands/list')

jest.mock('../../../src/shared')

jest.mock('../base')

describe('List Command', () => {
    let result

    beforeEach(() => {
        result = []
        jest.spyOn(process.stdout, 'write').mockImplementation((val) =>
            result.push(require('strip-ansi')(val.toString('utf8')))
        )
    })

    afterEach(() => jest.restoreAllMocks())

    test.each([[], ['--no-format']])('with flags:', async (...args) => {
        await ListCommand.run(args)
        expect(result).toMatchSnapshot()
    })
})
