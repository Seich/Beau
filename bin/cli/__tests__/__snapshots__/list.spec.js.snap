// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Command with flags: 1`] = `
Array [
  "  HTTP Verb           Alias                         Endpoint            
",
  "  GET                 alias                         https://example.org/anything
",
  "  GET                 teapot                        https://example.org/status/418
",
  "
",
]
`;

exports[`List Command with flags: 2`] = `
Array [
  "GET	alias	https://example.org/anything
",
  "GET	teapot	https://example.org/status/418
",
]
`;
